vformer/attention/vanilla.py

```diff
@@ -4,7 +4,19 @@
 
 
 class VanillaSelfAttention(nn.Module):
-    def __init__(self, dim, heads=8, dim_head=64, p_dropout=0.0):
+    """
+    class VanillaSelfAttention - Self attention mechanism for vanilla transformer
+
+    Inputs:
+    -----------------
+    dim: embeding dimension
+    heads: number of heads in attention
+    dim_head: dimension of head
+    p_dropout:probability for dropout layer
+    ----------------
+
+    """
+    def __init__(self, dim:int , heads:int=8, dim_head:int=64, p_dropout:float=0.0):
```

Comment:
Let's avoid type annotations for now.
```suggestion
    def __init__(self, dim , heads=8, dim_head=64, p_dropout=0.0):
```